client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java

```diff
@@ -71,35 +71,38 @@
     private boolean metaHeaderEnabled = true;
 
     static {
-        String version = ""; // unknown values are reported as empty strings in X-Elastic-Client-Meta
-        final CodeSource codeSource = RestClientBuilder.class.getProtectionDomain().getCodeSource();
-        if (codeSource != null) {
-            URL url = codeSource.getLocation();
-            if (url != null && url.toString().endsWith(".jar")) {
-                try (JarInputStream jar = new JarInputStream(url.openStream())) {
-                    Manifest manifest = jar.getManifest();
-                    String esVersion = manifest.getMainAttributes().getValue("X-Compile-Elasticsearch-Version");
-                    if (esVersion != null) {
-                        version = esVersion;
-                    }
-                } catch (Exception e) {
-                    // Keep version unknown
-                }
+
+        // Never fail on unknown version, even if an environment messed up their classpath enough that we can't find it.
+        // Better have incomplete telemetry than crashing user applications.
+        String version = null;
```

Comment:
nit: if we initialise it here with `""` as a default value, the below `if (version == null)`check is not needed anymore

Reply:
Actually it is: `versions.getProperty("elasticsearch-client")` can return `null` if the property file exists but doesn't contain that property. I agree that this is unlikely to happen, but client applications are a wild environment 😛 

So I finally chose that approach so that we don't have a mix of tests for `""` and `null`.

Reply:
Ah, good shout!